src/commands/Configuration/configure/automod/add.ts

```diff
@@ -117,17 +118,62 @@ async function add(interaction: MessageComponentInteraction, client: BulbBotClie
 
 					msgCollector?.on("collect", async (m: Message) => {
 						await m.delete();
-						if (selectedCategory && config[selectedCategory].includes(m.content)) {
+						let appendContent = m.content;
+						if (selectedCategory && config[selectedCategory].includes(appendContent)) {
 							await interaction.followUp({
 								content: await client.bulbutils.translate("config_automod_add_remove_add_already_exists", interaction.guild?.id, {
-									item: m.content,
+									item: appendContent,
 								}),
 								ephemeral: true,
 							});
 							return add(i, client, selectedCategory);
 						}
 
-						if (selectedCategory) await databaseManager.automodAppend(interaction.guild?.id as Snowflake, categories[selectedCategory], [m.content]);
+						// parsing of objects being done
+
+						// whitelist roles should only allow roles
+						if (selectedCategory === "ignoreRoles") {
+							const roles = appendContent.replace(NonDigits, "");
+							if (!i.guild?.roles.cache.get(roles)) {
+								await interaction.followUp({
+									content: await client.bulbutils.translate("config_automod_add_remove_add_invalid_role", interaction.guild?.id, {
+										item: appendContent,
+									}),
+									ephemeral: true,
+								});
+								return add(i, client, selectedCategory);
+							} else appendContent = roles;
+						}
+
+						// whitelist channels only allow channels
+						if (selectedCategory === "ignoreChannels") {
+							const channels = appendContent.replace(NonDigits, "");
+							if (!i.guild?.channels.cache.get(channels)) {
+								await interaction.followUp({
+									content: await client.bulbutils.translate("config_automod_add_remove_add_invalid_channel", interaction.guild?.id, {
+										item: appendContent,
+									}),
+									ephemeral: true,
+								});
+								return add(i, client, selectedCategory);
+							} else appendContent = channels;
+						}
+
+						// whitelist users only allow users ids
+						if (selectedCategory === "ignoreUsers") {
+							const users = appendContent.replace(NonDigits, "");
+							if (!i.guild?.members.cache.get(users)) {
+								await interaction.followUp({
+									content: await client.bulbutils.translate("config_automod_add_remove_add_invalid_user", interaction.guild?.id, {
+										item: appendContent,
+									}),
+									ephemeral: true,
+								});
+								return add(i, client, selectedCategory);
+							} else appendContent = users;
+						}
+
+						if (selectedCategory) await databaseManager.automodAppend(interaction.guild?.id as Snowflake, categories[selectedCategory], [appendContent]);
```

Comment:
```suggestion
						if (selectedCategory) await databaseManager.automodAppend(interaction.guild?.id, categories[selectedCategory], [appendContent]);
```
Nope, we aren't doing this anymore. I can see it was already there but it shouldn't have been. If the ID is actually required here, we need to be type guarding or we're just gonna have weird bugs. If this method doesn't actually require a non-nullish value, fix the type annotation so that it's not lying to us. If the annotation is already allowing a nullish value, we shouldn't be doing a type assertion for no reason. Note that `Snowflake` is just a direct alias for `string` so we can put `string`s into `Snowflake` parameters without having problems